tests/test_xpub_nodrop.cpp

```diff
@@ -54,14 +54,18 @@ void test ()
     int wait = 1;
     TEST_ASSERT_SUCCESS_ERRNO (zmq_setsockopt (pub, ZMQ_XPUB_NODROP, &wait, 4));
 
-
     //  Create a subscriber
     void *sub = test_context_socket (ZMQ_SUB);
     TEST_ASSERT_SUCCESS_ERRNO (zmq_connect (sub, "inproc://soname"));
 
     //  Subscribe for all messages.
     TEST_ASSERT_SUCCESS_ERRNO (zmq_setsockopt (sub, ZMQ_SUBSCRIBE, "", 0));
 
+    //  we must wait for the subscription to be processed here, so we just
+    //  hope SETTLE_TIME suffices; otherwise some or all published messages
+    //  might be lost
+    msleep (SETTLE_TIME);
```

Comment:
In these cases it's better to just switch to xpub and wait for the sub message directly. That way it's deterministic and cannot fail on slow platforms.

Reply:
Ah, that's even better, I wasn't aware of that possibility. :)

Reply:
Hm, the test_xpub_nodrop test case is still not running reliably indeed.

If we implement your suggestion, we are actually testing something different, since the publisher socket is the main test subject here. We would no longer test ZMQ_PUB but ZMQ_XPUB. (A test for ZMQ_XPUB with ZMQ_XPUB_NODROP set is actually missing at the moment).

Reply:
I don't think it's too much of an issue, pub is basically xpub with xrecv disabled

Reply:
ok, if it stays implemented that way. I will add a comment that this should be considered when making future changes

Reply:
yes I don't see that changing at any point, it's always been like that